### PR TITLE
Remove torrent file after download

### PR DESCRIPTION
**Pull request by gborri**

### CODE REVIEW

**Comment by fernandog** [comment]
Does it need to change DB version? We cant commit this now.

**Comment by fernandog** [comment]
Also can you check that tranmission error I send you?

**Comment by gborri** [comment]
@fernandog sent where? i can't find it

**Comment by fernandog** [comment]
@gborri 
https:// github .com /SiCKRAGETV/sickrage-issues/issues/679

**Comment by gborri** [comment]
@abeloin do you think is it possible to test and merge also this PR before merge to master? It has a db change.

**Comment by fernandog** [comment]
first develop>master
merge new features to develop
create experimental branch to test db changes.

**Comment by abeloin** [comment]
@gborri I won't be possible as we are now in release mode(ie: accepting bugfix and small feature only) to stabilize before releasing to master.

**Comment by fernandog** [comment]
@abeloin should we create an experimental branch? The problem is that users can switch to it. The db check must be in master before create the experimental branch

**Comment by abeloin** [comment]
@fernandog See your PM

**Comment by fernandog** [comment]
@gborri can you fully explain how this works?  we are discussing about it here: 
https://github.com/SiCKRAGETV/sickrage-issues/issues/911

**Comment by gborri** [comment]
@fernandog @abeloin dou you think ti PR could be merged?

**Comment by fernandog** [comment]
Not now. needs DB upgrade

**Comment by gborri** [comment]
@miigotu I don't want to change the InitialSchema and db versione. The philosophy (as i have understood) is to start from the actual InitialSchema and add following change by migration.
For your suggestion on code note: is not needed to check if is trasmission because if not the torrent_hash field will be empty.

**Comment by miigotu** [comment]
@gborri It is needed to update the InitialSchema when you change the database. The other check you already added is only supposed to be used when upgrading an existing database. Initial schema is on a fresh install, and should need no upgrades, so you have to modify the initial also.

**Comment by miigotu** [comment]
@gborri Also, you're right about not needing to check if its transmission, as you have a default 
def remove_torrent_downloaded(self,hash) in generic.py which just returns true, and that is properly done.

**Comment by gborri** [comment]
@miigotu ok, no problem for initial schema change. what i've said before is due to the fact that for a long period the initial schema was not changed. Now i've cheched and see that have update. 
for versione i can set 43 but when merged you need to check the correct one to set.

**Comment by gborri** [comment]
@miigotu Done.

**Comment by miigotu** [comment]
Ok, when I get around to making it possible to restore db to a lower version  backup when downgrading SR we will look into merging this.

**Comment by miigotu** [comment]
@gborri I like this idea, but I see two problems with it. It only handles transmission, and it should only try to remove from the client if the process method is 'move'. I would like to have this ready and on the list for the next db change.

**Comment by gborri** [comment]
@miigotu i handle only transmission because is the only client that I know. We can start from it and if there is someone with more knowledge about other client we can integrate it. for the "move" part i have not understood. the remove is in the postprocess procedure and i haven't seen anything related to the operation. If you give me any highlights i will check.

**Comment by miigotu** [comment]
If someone has the pp method set to copy or hard link or w/e they dont want it removed from their torrent client (they are seeding it)

If it is set to move, they cannot seed it regardless, so it should be removed without question, as the torrent client just gives an error about missing data.

This does not need to be configurable if made this way, we already know if a user wants to keep seeding based on the post processing procedure.

**Comment by gborri** [comment]
ok i git it. the point is the function of removing torrent need to be available only if process metod is set to move. In other cases the user have to remove it manually, correct?

**Comment by miigotu** [comment]
And it doesn't need to have any user configuration or options, just do it automatically. There shouldnt even be a gui checkbox.

File: sickbeard/postProcessor.py [hunk 1]
```diff
@@ -957,6 +958,15 @@ def process(self):
                 else:
                     cur_ep.release_name = ""
 
+                if cur_ep.torrent_hash != '':
+                    client = clients.getClientIstance(sickbeard.TORRENT_METHOD)()
```
Comment (miigotu) [inline sickbeard/postProcessor.py]: why trailing () ?

File: sickbeard/webserve.py [hunk 2]
```diff
@@ -1688,6 +1688,10 @@ def setStatus(self, show=None, eps=None, status=None, direct=False):
         if eps is not None:
 
             sql_l = []
+
+            if int(status) == SKIPPED:
+                client = clients.getClientIstance(sickbeard.TORRENT_METHOD)()
```
Comment (miigotu) [inline sickbeard/webserve.py]: trailing () ?

File: sickbeard/webserve.py [hunk 3]
```diff
@@ -1738,6 +1742,10 @@ def setStatus(self, show=None, eps=None, status=None, direct=False):
 
                     trakt_data.append((epObj.season, epObj.episode))
 
+                    if int(status) == SKIPPED and epObj.torrent_hash != '':
```
Comment (miigotu) [inline sickbeard/webserve.py]: something is not right here.

File: sickbeard/webserve.py [hunk 4]
```diff
@@ -1688,6 +1688,10 @@ def setStatus(self, show=None, eps=None, status=None, direct=False):
         if eps is not None:
 
             sql_l = []
+
+            if int(status) == SKIPPED:
```
Comment (miigotu) [inline sickbeard/webserve.py]: and here, something is not right.

File: sickbeard/postProcessor.py (hunk 1 above)
Comment (gborri) [inline sickbeard/postProcessor.py]: because if for any reason the value cur_ep.torrent_hash i don't want to call transmission

Comment (gborri) [inline sickbeard/postProcessor.py]: i don't remember. probably i have get this peace of code from other part of sickrage....

File: sickbeard/webserve.py (hunk 2 above)
Comment (gborri) [inline sickbeard/webserve.py]: same answear

File: sickbeard/webserve.py (hunk 3 above)
Comment (gborri) [inline sickbeard/webserve.py]: the point here is: if you change status of episode to default, you have to remove torrent from transmission and clean torrent_hash

File: sickbeard/webserve.py (hunk 4 above)
Comment (gborri) [inline sickbeard/webserve.py]: Why do you think is not right?

**Comment by miigotu** [comment]
Default is not always skipped.

It should only be removed on successful PP or marked as failed.

**Comment by matigonkas** [comment]
Just leaving my opinion ,i would prefer if it where configurable

**Comment by gborri** [comment]
@miigotu the line you have commented is on webserver.py. Here i will check if you want to change status to SKIPPED and in this case i will remove torrent from client and clean torrent hash.

**Comment by gborri** [comment]
@matigonkas ok

**Comment by fernandog** [comment]
I remove mine after download finished. Using script in transmission

**Comment by miigotu** [comment]
@matigonkas If your PP method is "Move", and the file is moved, why leave it in the client? You just get errors in your torrent client.

**Comment by matigonkas** [comment]
@miigotu its to keep track of what specific torrents i have downloaded,and the total nr

**Comment by gborri** [comment]
@matigonkas this information is on SickRage..... or if you have trakt enable also on it.

**Comment by gborri** [comment]
@miigotu is there a reason why have you closed this PR?

**Comment by miigotu** [comment]
16 commits, out of date, db version change, unmergable pr cleaning.

it is possible that this may be re-opened in the future, after we do the next db upgrade.  The next db upgrade will be planned with several db changes in one version so that hopefully the need for a db upgrade will not be needed for a very long time. Hash will be in the next version bump after the chnges have been decided and finalized.

**Pull request closed without merging**